src/store/plugins/obs/module/transitions.js

```diff
@@ -0,0 +1,59 @@
+export default {
+	state: {
+		current: null,
+		list: [],
+		duration: null
+	},
+	actions: {
+		'connection/closed'({commit}) {
+			commit('transitions/reset')
+		},
+		async 'connection/ready'({dispatch}) {
+			return dispatch('transitions/reload')
+		},
+		async 'transitions/reload'({commit, getters: {client}}) {
+			const {'current-transition': current, transitions} = await client.send({'request-type': 'GetTransitionList'})
+
+			commit('transitions/list', {transitions})
+			commit('transitions/current', {
+				'current-transition': current
+			})
+		},
+		'transitions/current'({getters: {client}}, {name}) {
+			return client.send({'request-type': 'SetCurrentTransition', 'transition-name': name})
+		},
+		'transitions/duration'({getters: {client}}, {ms}) {
+			return client.send({'request-type': 'SetTransitionDuration', 'duration': ms})
+		},
+		'event/SwitchTransition'({dispatch}) {
+			return dispatch('transitions/reload')
```

Comment:
use the transition name from event here and skip extra reload?

Reply:
Yes that needs to be done, I am a newb at this event handling and all. Learned (crash course) Vue.js Friday night and Saturday morning. --- the PAINT is still wet!